tests/Unit/Domain/Creators/Test_Sphere.cpp

```diff
@@ -56,147 +58,189 @@ std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>
+      // face, keep default
+
+      // anywhere on upper zeta face b/c adjacent with spherical wedge
+      if (rand_int_zeta == 1) {
+        xi_logical_coord = rand_real_xi;
+        eta_logical_coord = rand_real_eta;
+      }
+    } else {
+      // adjacent to wedges
+      // everywhere on low or high face should lie on a sphere b/c
+      // neighbor with spherical wedge
+      xi_logical_coord = rand_real_xi;
+      eta_logical_coord = rand_real_eta;
+    }
+
+    tnsr::I<double, 3, Frame::BlockLogical> block_logical_coords{
+        {{xi_logical_coord, eta_logical_coord, zeta_logical_coord}}};
+
+    double corner_distance_from_origin = 0.0;
+    // use stationary map if independent of time
+    if (time_dependencies.size() == 0) {
+      auto location_time_indep =
+          blocks[block_id]
+              .stationary_map()(block_logical_coords, 0.0)
+              .get_vector_of_data()
+              .second;
+      corner_distance_from_origin =
+          sqrt(square(location_time_indep[0]) + square(location_time_indep[1]) +
+               square(location_time_indep[2]));
+    } else {  // use if time dependent
+
+      for (auto& time_map_entry : initial_expiration_times) {
+        current_time = time_map_entry.second;
+        // go from logical to grid coords, then grid to intertial coords
+
+        auto inertial_location_time_dep =
+            blocks[block_id]
+                .moving_mesh_grid_to_inertial_map()(
+                    blocks[block_id].moving_mesh_logical_to_grid_map()(
+                        block_logical_coords, current_time, func_of_time),
+                    current_time, func_of_time)
+                .get_vector_of_data()
+                .second;
+
+        // origin in intertial frame (need to shift inertial
+        // coord by velocity * (final_time - initial_time))
+        corner_distance_from_origin =
+            sqrt(square(inertial_location_time_dep[0] -
+                        velocity[0] * (current_time - 1.0)) +
+                 square(inertial_location_time_dep[1] -
+                        velocity[1] * (current_time - 1.0)) +
+                 square(inertial_location_time_dep[2] -
+                        velocity[2] * (current_time - 1.0)));
+      }
+    }
+
+    // construct vector of inner radius, outer radius, and refinements levels
+    // where inertial block corners have to be located
+    std::vector<double> expected_corner_radii = radial_partitioning;
+    expected_corner_radii.insert(expected_corner_radii.begin(), inner_radius);
+    expected_corner_radii.emplace_back(outer_radius);
+
+    bool corner_matches_demarcations = false;
+    // loop through expected radii to verify at least 1 matches
+    for (const double radius : expected_corner_radii) {
+      if (corner_distance_from_origin == approx(radius)) {
+        corner_matches_demarcations = (corner_matches_demarcations or true);
+      }
+    }
+    CHECK(corner_matches_demarcations);
```

Comment:
(optional) you can use `std::any_of` or our simplified `alg::any_of` (in Utilities/Algorithm.hpp) here